docs/torch_bert_tutorial.md

```diff
@@ -0,0 +1,313 @@
+    I really like the new design of your website!
+    	label: 5 stars, with score: 0.6289
+    I’m not sure if I like the new design.
+    	label: 3 stars, with score: 0.5344
+    The new design is awful!
+    	label: 1 star, with score: 0.8435
+    It would be awesome if you give us feedback!
+    	label: 5 stars, with score: 0.4164
+
+
+Looks good! Next turn on mix-precision configuration `torch_config.enable_mlir_amp = True` and try it. Wish you have a good time.
+
+### Deep dive
+
+To have a glance at what BladeDISC have done during the optimization, one could print code of the forward
+method of the optimized module with the following example:
+
+
+```python
+# print the optimized code
+print(optimized_ts.code)
+```
+
+    def forward(self,
+        input_ids: Tensor,
+        attention_mask: Tensor,
+        input: Tensor) -> Dict[str, Tensor]:
+      seq_length = ops.prim.NumToTensor(torch.size(input_ids, 1))
+      _0 = torch.add(seq_length, CONSTANTS.c0, alpha=1)
+      _1 = torch.tensor(int(_0), dtype=None, device=None, requires_grad=False)
+      _2 = self.disc_grp0_len1264_0
+      _3 = [input, input_ids, attention_mask, _1]
+      _4, = (_2).forward(_3, )
+      return {"logits": _4}
+    
+
+
+As the printed code shows, almost the whole forward computations were clustered and lowered to BladeDISC
+(aka, the cluster `disc_grpx_lenxxx_x` where the optimizations happened).
+
+Another debugging tool is to set the environment variable `TORCH_ADDONS_DEBUG_LOG=on`. With the env variable set,
+the TorchScript subgraphs, their MHLO module lowered, and the compilation logs would be dumped and saved to local
+directory dump_dir. Since the dumped files grow very faster if the Module is large enough, we would like to choose
+a tiny DNN to demostrate the process:
+
+
+
+```python
+import os
+# set TORCH_ADDONS_DEBUG_LOG=on
+os.environ["TORCH_ADDONS_DEBUG_LOG"] = "on"
+# do BladeDISC optimization
+w = h = 10
+dnn = torch.nn.Sequential(
+            torch.nn.Linear(w, h),
+            torch.nn.ReLU(),
+            torch.nn.Linear(h, w),
+            torch.nn.ReLU()).cuda().eval()
+with torch.no_grad():
+    # BladeDISC torch_addons optimize will return an optimized TorchScript
+    opt_dnn_ts = torch_addons.optimize.optimize(
+        dnn, allow_tracing=True, model_inputs=(torch.ones(w, h).cuda(),))
+
+# print optimized code
+print(opt_dnn_ts.code)
+
+# list the debug files dumped
+!ls dump_dir
+```
+
+    def forward(self,
+        input: Tensor) -> Tensor:
+      _0 = (self.disc_grp0_len10_0).forward([input], )
+      _1, = _0
+      return _1
+    
+    dump.2021_12_22-18_24_04.226587.mlir
+    dump.2021_12_22-18_24_04.226587.pretty.mlir
+    graph.2021_12_22-18_24_04.226587.txt
+    mhlo_compile.2021_12_22-18_24_04.226587.log
+    out.2021_12_22-18_24_04.226587.so
+    out.2021_12_22-18_24_04.226587.so.pbtxt
+
+
+Those debug information would be useful when you run into problem. If the optimization isn't peforming as expected,
```

Comment:
"peforming"==>"performing"... better to use an automatic tool to help with the spell correction.